### DISCUSSION TITLE
Modifica di una proforma già creata

### DISCUSSION THREAD

**Discussion by dmigliacci** [fattureincloud/api#295] [Q&A] [answered]
Buongiorno,
ho realizzato la creazione di una proforma (con annesso ddt) e funziona tutto liscio come l'olio.

Vorrei ora fare la parte di modifica, ho trovato la PUT che fa al caso mio:

_Modify Issued Document_
_**PUT** /c/{company_id}/issued_documents/{document_id}_

Domande:

1. ciò che invierò nel body della PUT, sostituirà completamente il documento che era stato precedentemente creato?
2. inviando un documento di tipo proforma, questo modificherà anche il DDT che era stato autogenerato in fase di creazione?

Grazie in anticipo!
Davide

**Comment by emajo** [comment, marked as answer]
Ciao @dmigliacci, le chiamate di modifica non vanno a sostituire completamente il documento originale, ma solo i campi passati nel body della chiamata vengono effettivamente sovrascritti, per cui, se non si vanno a modificare i campi relativi al DDT questo non viene modificato.

Grazie,
Emanuele.

**Reply by dmigliacci** [reply]
Ciao Emanuele,
Ok chiarissimo.

Lato mio, per comodità, andrò a rispedire l'intero oggetto ricreandolo ex novo proprio come per la creazione.

Quindi mi aspetto che cambierà tutto, ddt incluso

**Reply by emajo** [reply]
In realtà passando l'intero documento originale con le modifiche che devono essere applicate non dovrebbe cambiare nulla a parte i campi modificati, ddt compreso, il mio consiglio è comunque di fare qualche tentativo per vedere se il comportamento è quello atteso.

Grazie,
Emanuele.

**Reply by valmoz** [reply]
Buongiorno @dmigliacci possiamo chiudere la conversazione?
Grazie

**Reply by dmigliacci** [reply]
Non ho avuto modo di provarlo. 
Direi che la risposta è cmq chiara. Chiudiamo e in caso di problemi riapro

**Reply by dmigliacci** [reply]
Confermo che funziona perfettamente, aggiorna l'intero documento!
